File: csharp/Algorithms/Ex41.cs
namespace Algorithms
{
    using System.Linq;
    public class Ex41
    {
        public static int iterativeSum(int[] array)
        {
            int result = 0;
            foreach (var item in array)
            {
                result += item;
            }

            return result;
        }

        public static int recursiveSum(int[] array)
        {
            if (array.Length == 0)
            {
                return 0;
            }
            if (array.Length == 1)
            {
                return array[0];
            }

            return array[0] + recursiveSum(array.Skip(1).ToArray());
        }
    }
}

File: java/src/main/java/grokking/algorithms/Ex41.java
package grokking.algorithms;

import java.util.Arrays;

public class Ex41 {
    public static int iterativeSum(int[] array) {
        int result = 0;
        for (int item : array) {
            result += item;
        }

        return result;
    }

    public static int recursiveSum(int[] array) {
        if (array.length == 0) {
            return 0;
        }
        if (array.length == 1) {
            return array[0];
        }

        return array[0] + recursiveSum(Arrays.stream(array).skip(1).toArray());
    }
}